src/components/RestaurantDetailModal.ts

```diff
@@ -0,0 +1,109 @@
+      <span class="restaurant__distance text-body">캠퍼스부터 ${restaurant.distance}분 내</span>
+      <p class="restaurant__description text-body">${restaurant.description ?? ''}</p>
+      <a href=${restaurant.link ?? ''} target="blank">${restaurant.link ?? ''}</a>
+    </div>
+    <div class="button-container">
+      <button type="button" class="button button--secondary text-caption">삭제하기</button>
+      <button type="button" class="button button--primary text-caption">닫기</button>
+    </div>
+  </section>
+`;
+
+class RestaurantDetailModal implements Component {
+  $target: Element;
+  #restaurant?: Restaurant;
+
+  constructor(parent: HTMLElement) {
+    parent.insertAdjacentHTML('beforeend', this.template());
+    this.$target = parent.lastElementChild!;
+  }
+
+  template = () => `
+    <div class="modal">
+      <div class="modal-backdrop"></div>
+      <div class="modal-container"></div>
+    </div>
+  `;
+
+  setRestaurant = (restaurant: Restaurant) => {
+    this.#restaurant = restaurant;
+  };
+
+  render = () => {
+    this.$target.querySelector('.modal-container')?.replaceChildren();
+    this.$target
+      .querySelector('.modal-container')
+      ?.insertAdjacentHTML('beforeend', detailTemplate(this.#restaurant!));
+  };
+
+  show = () => {
+    this.$target.classList.add('modal--open');
+  };
+
+  hide = () => {
+    this.$target.classList.remove('modal--open');
+  };
+
+  setCloseModalHandler = () => {
+    this.$target.querySelector('.button--primary')?.addEventListener('click', this.hide);
+    this.$target.querySelector('.modal-backdrop')?.addEventListener('click', this.hide);
+
+    document.addEventListener(
+      'keydown',
+      (event) => (event.key === 'Escape' || event.key === 'Esc') && this.hide(),
+    );
+  };
+
+  setFavoriteButtonHandler = (handler: (name: string) => void) => {
+    this.$target.querySelector('.button--favorite')?.addEventListener('click', (event) => {
+      const parent = (event.target as HTMLButtonElement).closest(
+        '.restaurant-detail-container',
+      ) as HTMLElement;
+
+      this.changeFavoriteButtonImage(
+        parent.querySelector('img[alt="즐겨찾기 추가"]') as HTMLImageElement,
+      );
+
+      handler((parent.querySelector('.restaurant__name') as HTMLElement).innerText);
+    });
+  };
+
+  setDeleteButtonHandler = (handler: (name: string) => void) => {
+    this.$target.querySelector('.button--secondary')?.addEventListener('click', (event) => {
+      const parent = (event.target as HTMLButtonElement).closest(
+        '.restaurant-detail-container',
+      ) as HTMLElement;
+
+      handler((parent.querySelector('.restaurant__name') as HTMLElement).innerText);
+      this.hide();
+    });
+  };
+
+  private changeFavoriteButtonImage = (image: HTMLImageElement) => {
+    image.src === FavoriteIconImagePath.DEFALUT
+      ? (image.src = FavoriteIconImagePath.ADDED)
+      : (image.src = FavoriteIconImagePath.DEFALUT);
```

Comment:
```suggestion
 
    image.src = image.src === FavoriteIconImagePath.DEFALUT 
                         ? FavoriteIconImagePath.ADDED 
                         :  FavoriteIconImagePath.DEFALUT
```

삼항연산은 가능한 동작보단 값을 반환하도록 해보아요